### PR TITLE
[WIP] feat: Adapting the spring-ai to the TongYI model

### PR DESCRIPTION
**Pull request by yuluo-yx**
### Describe what this PR does / why we need it

- For `spring-cloud-alibaba-ai`.
- Completed chat completion API adaptation.

[Spring AI](https://docs.spring.io/spring-ai/reference/0.8-SNAPSHOT/index.html)
[TongYi models](https://help.aliyun.com/zh/dashscope/developer-reference/model-introduction)
[Spring AI chat completion api](https://docs.spring.io/spring-ai/reference/0.8-SNAPSHOT/api/chatclient.html)

### CODE REVIEW

File: spring-cloud-alibaba-dependencies/pom.xml [hunk 1]
```diff
@@ -198,6 +203,24 @@
...
+                <groupId>com.alibaba</groupId>
+                <artifactId>dashscope-sdk-java</artifactId>
+                <version>${dashscope-sdk-java.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>org.springframework.ai</groupId>
+                <artifactId>spring-ai-core</artifactId>
+                <version>${spring.ai.version}</version>
+            </dependency>
```
Comment (ruansheng8) [inline spring-cloud-alibaba-dependencies/pom.xml]: 这部分依赖需要放到上面，Own dependencies - Starters 里面只放 sca starter 模块

File: spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-ai/src/main/java/com/alibaba/cloud/ai/tongyi/aot/TongYiRuntimeHints.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+package com.alibaba.cloud.ai.tongyi.aot;
+
+import org.springframework.aot.hint.RuntimeHints;
+import org.springframework.aot.hint.RuntimeHintsRegistrar;
+
+/**
+ * The TongYiRuntimeHints class is responsible for registering runtime hints for TongYiAI.
+ *
+ * @author yuluo
+ * @author 1481556636@qq.com
```
Comment (ruansheng8) [inline spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-ai/src/main/java/com/alibaba/cloud/ai/tongyi/aot/TongYiRuntimeHints.java]: 格式有误

File: spring-cloud-alibaba-dependencies/pom.xml (hunk 1 above)
Comment (yuluo-yx) [reply spring-cloud-alibaba-dependencies/pom.xml]: done!

File: spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-ai/src/main/java/com/alibaba/cloud/ai/tongyi/aot/TongYiRuntimeHints.java (hunk 2 above)
Comment (yuluo-yx) [reply spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-ai/src/main/java/com/alibaba/cloud/ai/tongyi/aot/TongYiRuntimeHints.java]: done!

**Review by chickenlj — APPROVED** [review]
LGTM.

**Pull request merged**